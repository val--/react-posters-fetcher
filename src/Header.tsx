import { Button, CircularProgress, makeStyles } from '@material-ui/core';

import { NavLink } from 'react-router-dom';

import { useLoading } from './store';

const Header = () => {
    const classes = useStyles();
    const loading = useLoading();
    return (
        <header>
            <NavLink
                style={{ textDecoration: 'none' }}
                activeStyle={{ textDecoration: 'underline' }}
                to="/posters"
            >
                <Button>Posters</Button>
            </NavLink>{' '}
            <NavLink
                style={{ textDecoration: 'none' }}
                activeStyle={{ textDecoration: 'underline' }}
                to="/customers"
            >
                <Button>Customers</Button>
            </NavLink>
            <div className={classes.loader}>
                <div style={{ display: loading ? 'block' : 'none' }}>
                    <CircularProgress size={16} />
                </div>
            </div>
        </header>
    );
};

const useStyles = makeStyles((theme) => ({
    loader: {
        display: 'inline-block',
        height: 16,
        width: 16,
    },
}));

export default Header;
