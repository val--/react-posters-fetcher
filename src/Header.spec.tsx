import { FunctionComponent } from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

import Header from './Header';

jest.mock('./store', () => ({
    useLoading: () => false,
}));
const AppMock: FunctionComponent = ({ children }) => <BrowserRouter>{children}</BrowserRouter>;

test('Header should show loading when loading', () => {
    const component = renderer.create(
        <AppMock>
            <Header />
        </AppMock>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

jest.mock('./store', () => ({
    useLoading: () => false,
}));

test('Header should not show loading when not loading', () => {
    const component = renderer.create(
        <AppMock>
            <Header />
        </AppMock>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
