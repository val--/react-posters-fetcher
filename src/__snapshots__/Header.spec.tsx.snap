// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should not show loading when not loading 1`] = `
<header>
  <a
    aria-current={null}
    href="/posters"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Posters
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </a>
   
  <a
    aria-current={null}
    href="/customers"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Customers
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </a>
  <div
    className="makeStyles-loader-1"
  >
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 16,
            "width": 16,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</header>
`;

exports[`Header should show loading when loading 1`] = `
<header>
  <a
    aria-current={null}
    href="/posters"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Posters
      </span>
    </button>
  </a>
   
  <a
    aria-current={null}
    href="/customers"
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Customers
      </span>
    </button>
  </a>
  <div
    className="makeStyles-loader-1"
  >
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 16,
            "width": 16,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</header>
`;
